In [1]:
import pandas as pd

veri = pd.read_csv("C:\\Users\\cdggt\\Desktop\\country.csv")

In [2]:
nufusazalan = veri.sort_values(by="Population", ascending=False)
print("Nüfusa göre azalan :")
print(nufusazalan[["Country", "Population"]])

Nüfusa göre azalan :
                   Country  Population
42                  China   1313973713
94                  India   1095351995
214         United States    298444215
95              Indonesia    245452739
27                 Brazil    188078227
..                     ...         ...
144                 Nauru        13287
209                Tuvalu        11810
140            Montserrat         9439
171          Saint Helena         7502
174  St Pierre & Miquelon         7026

[227 rows x 2 columns]


Ülkeler nufusa göre aazalan sıralandığında en kalabalık ülke Çin olurken en az nufusa sahip ülke St Pierre & Miquelon

In [3]:
gdpmiktarı= veri.sort_values(by="GDP ($ per capita)", ascending=True)
print("\nKişi basına düşen milli hasılaya göre artan sıralama:")
print(gdpmiktarı[["Country", "GDP ($ per capita)"]])


Kişi basına düşen milli hasılaya göre artan sıralama:
             Country  GDP ($ per capita)
58       East Timor                500.0
183    Sierra Leone                500.0
188         Somalia                500.0
33          Burundi                600.0
125          Malawi                600.0
..               ...                 ...
22          Bermuda              36000.0
154          Norway              37800.0
214   United States              37800.0
121      Luxembourg              55100.0
223  Western Sahara                  NaN

[227 rows x 2 columns]


En düşük kişi başı düşen yıllık gelire Batı Timor(500$) sahipken , En yüksek gelire Lüksemburg(55100$) sahip.

In [4]:
insansayısı = veri[veri["Population"] > 10_000_000]
print("\nNüfusu 10 milyonun üzerinde olan ülkeler:")
print(insansayısı[["Country", "Population"]])


Nüfusu 10 milyonun üzerinde olan ülkeler:
          Country  Population
0    Afghanistan     31056997
2        Algeria     32930091
5         Angola     12127071
8      Argentina     39921833
11     Australia     20264082
..            ...         ...
218    Venezuela     25730435
219      Vietnam     84402966
224        Yemen     21456188
225       Zambia     11502010
226     Zimbabwe     12236805

[79 rows x 2 columns]


Popülasyonu 10 milyon kişi üzerinde olan toplam 79 ülke bulunmaktadır.

In [5]:
OkurYazarlık = veri.sort_values(by="Literacy (%)", ascending=False).head(5)
print("\nEn yüksek Okur yazarlık oranına sahip 5 ülke:")
print(OkurYazarlık[["Country", "Literacy (%)"]])


En yüksek Okur yazarlık oranına sahip 5 ülke:
             Country Literacy (%)
53   Czech Republic          99,9
93          Iceland          99,9
163          Poland          99,8
114          Latvia          99,8
64          Estonia          99,8


Okur yazarlık oranında avrupa ve iskandinav ülkelerinin ilk beşte olduğu görünüyor

In [6]:
kazanc10000 = veri[veri["GDP ($ per capita)"] > 10000]
print("\nKişi basına yıllık 10.000$\'dan yüksek olan ülkeler:")
print(kazanc10000[["Country", "GDP ($ per capita)"]])


Kişi basına yıllık 10.000$'dan yüksek olan ülkeler:
                   Country  GDP ($ per capita)
4                 Andorra              19000.0
7       Antigua & Barbuda              11000.0
8               Argentina              11200.0
10                  Aruba              28000.0
11              Australia              29000.0
..                     ...                 ...
212  United Arab Emirates              23200.0
213        United Kingdom              27700.0
214         United States              37800.0
215               Uruguay              12800.0
220        Virgin Islands              17200.0

[76 rows x 2 columns]


Dünya genelinde bu kriterlerde 76 ülke bulunmaktadır

In [7]:
veri["Population Density"] = veri["Population"] / veri["Area (sq. mi.)"]
AlanBasıİnsan= veri.sort_values(by="Population Density", ascending=False).head(10)
print("\n1 mil kare alana en çok insan düşen 10 ülke:")
print(AlanBasıİnsan[["Country", "Population Density"]])


1 mil kare alana en çok insan düşen 10 ülke:
         Country  Population Density
138      Monaco         16271.500000
122       Macau         16183.035714
184   Singapore          6482.178932
91    Hong Kong          6355.706960
78    Gibraltar          3989.714286
74   Gaza Strip          3968.769444
129       Malta          1266.500000
22      Bermuda          1241.000000
127    Maldives          1196.693333
15      Bahrain          1050.503759


1 mil kare basına en yüksek popülasyona sahip ülke 16271.5 kişi ile Monaco.